### PR TITLE
Move some pythran config from CI to setup

### PR DESCRIPTION
**Pull request by serge-sans-paille**
It's more reproducible and less configuration hassle that way.
Somehow related to #13464

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @serge-sans-paille. The CI failure on 32-bit Windows caught an issue with the Pythran implementation of `_group_columns` it looks like.

**Comment by serge-sans-paille** [comment]
Looks like distutils on Windows doesn't honor CXXFLAGS=-m32 or also requires LDFLAGS=-m32. I reverted that part which is not directly linked to that commit main goal.

File: scipy/signal/setup.py [hunk 1]
```diff
@@ -30,7 +30,8 @@ def configuration(parent_package='', top_path=None):
```
Comment (rgommers) [inline scipy/signal/setup.py:30]: There's a second `PythranExtension` in this file, for `_max_len_seq_inner`. Could you add `config=` there too?

Other than that, this looks good to me.

**Comment by rgommers** [comment]
Merged, thanks @serge-sans-paille!

**Pull request merged**